python/hail/docs/scan.rst

```diff
@@ -0,0 +1,18 @@
+.. _sec-scan:
+
+Scans
+===========
+
+The ``scan`` module is exposed as ``hl.scan``, e.g. ``hl.scan.sum``.
+
+The functions in this module perform rolling aggregations along the desired axis;
```

Comment:
Users are going to need explanation to make sense of what this does. It's nice to mention that scan.count is equivalent to add_[row/col]_index, but I think also showing concrete examples with `sum` on a table and then on a matrix table would help a lot. Note that the order comes from the row / column order.

I'm worried that some of these (like hardy_weinberg) have extremely expensive `result` functions. There should at least be a warning